### PR TITLE
Introduce the ability to specify an IdGenerator in the TracerFactory configuration

### PR DESCRIPTION
**Pull request by amcghie**
Also provide the ability to customize the `ServerTracingDynamicFeature` via the `HaystackTracerBundle` constructor.

### CODE REVIEW

**Comment by amcghie** [comment]
@doctorXWrites I'm not sure you brought my changes into your PR.  Would you mind reviewing this PR please.

File: src/main/java/com/expedia/haystack/dropwizard/configuration/TracerFactory.java [hunk 1]
```diff
@@ -53,6 +51,9 @@
     @NotEmpty
     private List<DispatcherFactory> dispatchers = ImmutableList.of(new RemoteDispatcherFactory());
 
+    @Valid
+    private IdGenerator idGenerator;
```
Comment (mchandramouli) [inline src/main/java/com/expedia/haystack/dropwizard/configuration/TracerFactory.java:55]: Can we add LongIdGenerator as default?

```private IdGenerator idGenerator = new LongIdGenerator()```

**Review by mchandramouli — CHANGES_REQUESTED** [review]

**Comment by mchandramouli** [comment]
@amcghie  reviewed your PR. Looks good. Thanks for raising this PR. I left one minor comment. Can you look at it? I will merge once we discuss/resolve that this morning

File: src/main/java/com/expedia/haystack/dropwizard/configuration/TracerFactory.java (hunk 1 above)
Comment (amcghie) [reply src/main/java/com/expedia/haystack/dropwizard/configuration/TracerFactory.java:55]: Hi @mchandramouli, I did consider giving this a default value but decided against it as the `Tracer` (that's being built by this factory) already has a sensible default if an `idGenerator` is not provided (or is null).  By adding another default here I was concerned that the two default values may get out of sync.

Happy to implement your suggestion, just thought i'd explain my rationale for not providing one.

Comment (mchandramouli) [reply src/main/java/com/expedia/haystack/dropwizard/configuration/TracerFactory.java:55]: @amcghie good point. I am resolving this comment and merging your PR. when you get a chance can you issue a PR for README

**Review by mchandramouli — APPROVED** [review]

**Comment by mchandramouli** [comment]
@amcghie thanks for your contribution. We highly appreciate it

**Pull request merged**